lib/assets/javascripts/views/usage/index.js

```diff
@@ -45,35 +58,51 @@ $(() => {
 
     return rangeDates;
   };
-  const initialise = () => {
-    $.ajax({
-      headers: { Authorization: `Token token="${apiToken}"` },
-      url: topicToURL.users,
-      data: { range_dates: rangeDatesUpToLastYearFromNow() },
-    }).then((data) => {
-      new Chart($('#yearly_users'), { // eslint-disable-line no-new
-        type: 'bar',
-        data: {
-          labels: Object.keys(data),
-          datasets: [{
-            data: Object.keys(data).map(k => data[k]),
-            backgroundColor: '#4F5253', // TODO parameterised according to roadmap main colour instance
-          }],
+  const createChart = ({ selector, data, appendTolabel = '' } = {}) => {
+    new Chart($(selector), { // eslint-disable-line no-new
+      type: 'bar',
+      data: {
+        labels: Object.keys(data),
+        datasets: [{
+          data: Object.keys(data).map(k => data[k]),
+          backgroundColor: '#4F5253',
+          // TODO parameterised according to roadmap main colour instance
+        }],
+      },
+      options: {
+        legend: {
+          display: false,
         },
-        options: {
-          legend: {
-            display: false,
-          },
-          tooltips: {
-            callbacks: {
-              label: tooltipItem => `${tooltipItem.yLabel} users`,
-            },
+        tooltips: {
+          callbacks: {
+            label: tooltipItem => `${tooltipItem.yLabel} ${appendTolabel}`,
           },
         },
-      });
-    }, (jqXHR) => {
-      console.log('error: %o', jqXHR);
+      },
     });
   };
+  // Sends an AJAX request to our two current endpoints that generate yearly data
+  // (e.g. users_joined_api_v0_statistics_path, completed_plans_api_v0_statistics_path )
+  // and draws a barChart when success response is found
+  const initialise = () => {
+    // Only fire AJAX requests if topicToURL object has keys, i.e. topics mapping to URLs
+    if (Object.keys(topicToURL).length > 0) {
```

Comment:
Thank you. Initialize was being fired on every page before and causing JS errors. This should fix that